api/views.py

```diff
@@ -50,3 +50,21 @@ def destroy(self, request, pk=None, **kwargs):
             puzzle.delete()
 
         return Response({})
+
+    def partial_update(self, request, pk=None, **kwargs):
+        puzzle = None
+        with transaction.atomic():
+            puzzle = self.get_object()
+            serializer = self.get_serializer(puzzle, data=request.data, partial=True)
+            serializer.is_valid(raise_exception=True)
+            data = serializer.validated_data
+            puzzle.update_metadata(
```

Comment:
I think `update_metadata()` already calls save()?

Reply:
Yeah, but the call on line 68 is for saving the change on line 67 when we're changing the puzzle status (stuck/extraction/etc), which is not handled by update_metadata. You could probably refactor this so that when we're updating both eg the puzzle name and the status at the same time (not supported by the UI currently but supported by the API) we only make one call to `.save()` instead of two but for now I'm not worried about it

Reply:
Hmm... it seems a little strange to have two save paths: puzzzle.update_metadata() and puzzle.save(). Any chance these can be combined? Also, I see that update_metadata() calls update_sheets() on commit. When status changes (e.g.: we have a new confirmed answer for a meta) and puzzle.save() is called, is update_sheets() also called?

Reply:
Yeah, I agree. For the moment I just wanted to focus on porting over the front-end and the existing structure as much as possible without also rewriting the model code, but I'd like to make this part better too. Ideally I think we get rid of puzzle.update_metadata and move some of the logic elsewhere (inside puzzle.save(), or pre/post_save hooks, or checks in the view methods).

As for status changes, by themselves right now they don't affect the sheets (eg a puzzle moving from SOLVING -> STUCK). Adding an answer isn't supported yet but when it is I'll make sure it updates the sheets.

Reply:
I filed #271 for updating sheets when updating answers so we don't forget